src/spdx_matcher/__init__.py

```diff
@@ -626,6 +634,38 @@ def _load_license_analyser_cache():
     return index, match_cache
 
 
+def matcher_poster(func):
```

Comment:
Add decorator here instead of logic in method `analyse_license_text ` is because `analyse_license_text ` have multiple return syntax. And we have to avoid duplicate logic in codebase